jetbrains-core/src/software/aws/toolkits/jetbrains/services/s3/objectActions/NewFolderAction.kt

```diff
@@ -0,0 +1,44 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.services.s3.objectActions
+
+import com.intellij.openapi.actionSystem.AnActionEvent
+import com.intellij.openapi.ui.Messages
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.launch
+import software.aws.toolkits.jetbrains.components.telemetry.ActionButtonWrapper
+import software.aws.toolkits.jetbrains.services.s3.editor.S3TreeTable
+import software.aws.toolkits.jetbrains.services.s3.editor.S3VirtualBucket
+import software.aws.toolkits.jetbrains.services.s3.editor.getDirectoryKey
+import software.aws.toolkits.jetbrains.utils.notifyError
+import software.aws.toolkits.resources.message
+
+class NewFolderAction(
+    private val treeTable: S3TreeTable,
+    val bucket: S3VirtualBucket
+) : ActionButtonWrapper(message("s3.new.folder"), null, null) {
+
+    override fun isEnabled(): Boolean = treeTable.selectedRows.size <= 1
+
+    override fun doActionPerformed(e: AnActionEvent) {
+
+        val node = treeTable.selectedRows.firstOrNull()?.let { treeTable.getNodeForRow(it) } ?: treeTable.getRootNode()
+
+        Messages.showInputDialog(e.project, message("s3.new.folder.name"), message("s3.new.folder"), null)?.let { key ->
```

Comment:
Do we want to block `/` in the entered name?

Reply:
you mean just plain ol "/" or anything with a "/" in it?

Reply:
Both.

Should we allow foo/bar/file?
Or /foo/ seems odd as well

Reply:
I think we should leave the validation to S3, I wonder what /folder would do

Reply:
/folder "works" it just creates a blank node with a sub-folder of "folder"